cmd/provenanced/cmd/root.go

```diff
@@ -222,6 +222,7 @@ func queryCommand() *cobra.Command {
 	)
 
 	cmd.PersistentFlags().String(flags.FlagChainID, "", "The network chain ID")
+	cmd.PersistentFlags().String(flags.FlagNode, "","The node to connect to (IP:Port)")
```

Comment:
Generally speaking each command will have [`flags.AddQueryFlagsToCmd(cmd)`](https://github.com/cosmos/cosmos-sdk/blob/1b517b02b48197794c8162f4ac0f03bd8114db99/client/flags/flags.go#L105-L115) added (as appropriate).  As one example the [`server.QueryBlockCmd()`](https://github.com/cosmos/cosmos-sdk/blob/31ef899f4c3c091b641e68be25214715f4ea78c7/server/cmt_cmds.go#L302) already registers the required flag.

Reply:
Hi @iramiller 
Correct, it should have recognized the node flag but it did not. Should I add  [flags.AddQueryFlagsToCmd(cmd)](https://github.com/cosmos/cosmos-sdk/blob/1b517b02b48197794c8162f4ac0f03bd8114db99/client/flags/flags.go#L105-L115) here instead?

Reply:
As @SpicyLemon pointed out it shouldn’t need this flag since this is supposed to be a local only command. 

If the process is trying to make an out going connection then that seems to be the issue that needs to be fixed vs the node connection flag.

Reply:
Yes, the `build-address` command should have [flags.AddQueryFlagsToCmd(cmd)](https://github.com/cosmos/cosmos-sdk/blob/1b517b02b48197794c8162f4ac0f03bd8114db99/client/flags/flags.go#L105-L115) applied to it. That looks like an oversight by the wasm team.

To fix it in this repo, you'd have to do something similar to `fixTxWasmInstantiate2Aliases`. After all the sub-commands have been added, find the one in question and apply `AddQueryFlagsToCmd` to it.

Reply:
Hey, I tried the snippet below as suggested by @SpicyLemon, but it didn’t quite work. Could you please tell me if I'm missing something here?
```bash
func fixBuildAddressQueryFlags(rootCmd *cobra.Command) {
    // Find the "build-address" command.

    // cmd, _, err := rootCmd.Find([]string{"query", "build-address"})
    cmd, _, err := rootCmd.Find([]string{"query", "wasm",  "build-address"})
    if err != nil || cmd == nil {
        // If the command doesn't exist, there's nothing to do.
        return
    }

    // Apply query flags to the command.
    flags.AddQueryFlagsToCmd(cmd)
}
```
this is how the queryCommand function looked like
```bash
func queryCommand() *cobra.Command {
	cmd := &cobra.Command{
		Use:                        "query",
		Aliases:                    []string{"q"},
		Short:                      "Querying subcommands",
		DisableFlagParsing:         true,
		SuggestionsMinimumDistance: 2,
		RunE:                       client.ValidateCmd,
	}

	cmd.AddCommand(
		rpc.ValidatorCommand(),
		rpc.QueryEventForTxCmd(),
		server.QueryBlockCmd(),
		authcmd.QueryTxsByEventsCmd(),
		server.QueryBlocksCmd(),
		authcmd.QueryTxCmd(),
		server.QueryBlockResultsCmd(),
	)
	
	cmd.PersistentFlags().String(flags.FlagChainID, "", "The network chain ID")
	fixBuildAddressQueryFlags(cmd)
	// flags.AddQueryFlagsToCmd(cmd)

	return cmd
}

```

Reply:
At the point where you call `fixBuildAddressQueryFlags`, the query commands for each module have not yet been added. In `NewRootCmd`, there's a call to `fixTxWasmInstantiate2Aliases` that fixes one of the wasm tx commands. That command is added with the other module commands. So right after that would be a good place to call your fixer.

Also, can we call it `fixQueryWasmBuildAddressFlags`? That way it's named after the actual command and uses the same naming scheme as the other fixers in there.

Reply:
Hi @SpicyLemon 
Thank you very much for the input. That has resolved the issue.